CMakeLists.txt

```diff
@@ -107,11 +101,44 @@ if (EKAT_ENABLE_TESTS)
   add_subdirectory(tests)
 endif()
 
-# Generate configuration file
-include (EkatUtils)
-EkatConfigFile (${EKAT_SOURCE_DIR}/ekat_config.h.in
-                ${EKAT_BINARY_DIR}/ekat_config.h
-                F90_FILE ${EKAT_BINARY_DIR}/ekat_config.f)
-install(FILES ${EKAT_BINARY_DIR}/ekat_config.h
-              ${EKAT_BINARY_DIR}/ekat_config.f
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/include/ekat)
+###########################################
+###   Package ekat as a CMake package   ###
+###########################################
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+CONFIGURE_PACKAGE_CONFIG_FILE(
+  cmake/EkatConfig.cmake.in
+  "${EKAT_BINARY_DIR}/EkatConfig.cmake"
+  INSTALL_DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/cmake)
+
+WRITE_BASIC_PACKAGE_VERSION_FILE("${EKAT_BINARY_DIR}/EkatConfigVersion.cmake"
+    VERSION ${EKAT_VERSION_MAJOR}.${EKAT_VERSION_MINOR}.${EKAT_VERSION_PATCH}.
+    COMPATIBILITY SameMajorVersion)
+
+# Install the EkatConfig*.cmake files
+install(FILES
+  "${EKAT_BINARY_DIR}/EkatConfig.cmake"
+  "${EKAT_BINARY_DIR}/EkatConfigVersion.cmake"
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ekat)
+
+# Install cmake targets
+install(EXPORT EkatTargets
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ekat)
+
+###########################################
+###   Install ekat cmake util scripts   ###
+###########################################
+
+install (FILES
+  ${EKAT_CMAKE_PATH}/EkatCreateUnitTest.cmake
+  ${EKAT_CMAKE_PATH}/EkatSetCompilerFlags.cmake
+  ${EKAT_CMAKE_PATH}/EkatUtils.cmake
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ekat/cmake)
```

Comment:
An alternative could be `<prefix>/share/cmake`, but it already contains yamlcpp cmake config stuff (albeit in a yaml-cpp subfolder). We could modify yamlcpp (we are using our own fork, afterall), to install its cmake config sutff in lib64/cmake, just like ekat and kokkos (would be simmetrycal).

Reply:
The `share` path is the approach I've seen taken most often. I don't have strong feelings, though.

Regarding `yaml-cpp`--it's a much more general library than `ekat` or `kokkos`, so I don't mind it having a different set of conventions. If we ever want to merge our changes back into `yaml-cpp`, it might be in our interest to leave things like this alone.

Reply:
I also would like to use `share/cmake`. Maybe I'll stuff our cmake scripts in `share/cmake/Modules`, which mimics cmake installation folder.